services/meta/data.go

```diff
@@ -783,6 +784,91 @@ func (data *Data) hasAdminUser() bool {
 	return false
 }
 
+// ImportData imports selected data into the current metadata.
+// opts provides options to possibly rename the Database or Retention Policy, or change the Replication Factor.
+func (data *Data) ImportData(other Data, backupDBName, restoreDBName, backupRPName, restoreRPName string) (map[uint64]uint64, error) {
+
+	if backupDBName == "" {
+		// if no backupDBName then we'll try to import all the DB's.  If one of them fails, we'll mark the whole
+		// operation a failure and return an error.
+		shardIDMap := make(map[uint64]uint64)
+		for _, dbi := range other.Databases {
+			if dbi.Name == "_internal" {
+				continue
+			}
+			tmpMap, err := data.ImportOneDB(other, dbi.Name, "", "", "")
+			if err != nil {
+				return shardIDMap, err
+			}
+
+			for k, v := range tmpMap {
```

Comment:
Maybe just pass the shard map into the function, rather than copying all the entries around? Also, perhaps `ImportOneDB` should be unexported?

Reply:
👍